src/core_ocean/shared/mpas_ocn_vmix_gotm.F

```diff
@@ -0,0 +1,531 @@
+            gotmVertDiffTopOfCellNew(k, iCell) = gotm_vertDiffT(ki)
+            gotmTKETopOfCellNew(k, iCell) = gotm_TKE(ki)
+            gotmDissTopOfCellNew(k, iCell) = gotm_eps(ki)
+            gotmLengthTopOfCellNew(k, iCell) = gotm_L(ki)
+            gotmKbTopOfCellNew(k, iCell) = gotm_kb(ki)
+            gotmEpsbTopOfCellNew(k, iCell) = gotm_epsb(ki)
+            ! update viscosity/diffusivity
+            vertViscTopOfCell(k, iCell) = gotm_vertVisc(ki)
+            vertDiffTopOfCell(k, iCell) = gotm_vertDiffT(ki)
+         end do
+
+      end do
+      !$omp end do
+      !$omp end parallel
+
+      call mpas_timer_stop('gotm cell loop')
+
+      deallocate(shearSquaredTop)
+      deallocate(NSquaredTop)
+      deallocate(hLayer)
+
+   !--------------------------------------------------------------------
+
+   end subroutine ocn_vmix_coefs_gotm_build!}}}
+
+!***********************************************************************
+!
+!  routine ocn_vmix_gotm_init
+!
+!> \brief   Initializes ocean vertical mixing quantities using GOTM
+!> \author  Qing Li
+!> \date    11 May 2020
+!> \details
+!>  This routine initializes a variety of quantities related to
+!>  vertical mixing in the ocean. Parameters are set by calling into GOTM
+!
+!-----------------------------------------------------------------------
+
+
+   subroutine ocn_vmix_gotm_init(domain,err)!{{{
+
+   !--------------------------------------------------------------------
+
+      implicit none
+
+      type (domain_type), intent(inout) :: domain !< Input/Output: domain information
+
+      integer, intent(out) :: err !< Output: error flag
+
+      type (block_type), pointer :: block
+      type (mpas_pool_type), pointer :: meshPool
+      type (mpas_pool_type), pointer :: statePool
+
+      integer, pointer :: nVertLevels, nVertLevelsP1
+      integer, dimension(:), pointer :: nCellsArray
+
+      real (kind=RKIND), dimension(:,:), pointer :: &
+        gotmVertViscTopOfCellCur, gotmVertDiffTopOfCellCur, &
+        gotmTKETopOfCellCur, gotmDissTopOfCellCur, gotmLengthTopOfCellCur, &
+        gotmKbTopOfCellCur, gotmEpsbTopOfCellCur
+
+      ! TODO: read in namelist for testing <20200512, Qing Li> !
+      integer :: namlstUnit !< unit for GOTM namelist file
+      integer :: k, ki, iCell, nCells
+      type (MPAS_timeInterval_type) :: timeStep
+
+      !
+      ! assume no errors during initialization and set to 1 when error is encountered
+      !
+      err=0
+
+      gotmOn = config_use_gotm
+
+      !
+      ! only initialize if GOTM is turned on
+      !
+      if (.not. gotmOn) return
+
+      !
+      ! When GOTM is turned on, all other vertical mixing schemes should be off
+      ! Test to make sure this is the case.
+      !
+      ! test here, err=1 if a problem
+
+      ! TODO: use MPAS-O dt for testing. Sub-stepping is possible, though it may not help too much without updating the shear and stratification in the substeps <20200604, Qing Li> !
```

Comment:
@qingli411 please search for all lines with TODO. We want to either take care of it, or add it as a comment that says `comment, possible future work`. Committing TODO lines into the code base makes it look like we forgot something.

Reply:
@mark-petersen OK. I will do that.